framework/src/test/java/uk/gov/ons/ctp/common/cloud/RetryableCloudDataStoreSpringTest.java

```diff
@@ -32,6 +34,7 @@
       "cloud-storage.backoff.max=300",
       "cloud-storage.backoff.max-attempts=3",
     })
+@ResourceLock(value = "RetryableCloudDataStoreSpringTest", mode = READ_WRITE)
```

Comment:
As mentioned I am uncomfortable with the pattern of the key (value) name being the same as the class name in the case of using Spring-based tests.   I have not been able to demonstrate my expected problem here, but I have seen that Spring ApplicationContext is a global that spans multiple JUnit  classes , and therefore I believe there is a potential for intermittent failures and I would prefer the value be a constant used for all Spring based tests.  (or just use the `@Isolated` annotation).

To continue with the `@ResourceLock` annotation, I would create a global value in the framework and references it from all Spring based tests as so:
`
@ResourceLock(value = LockKey.SPRING_TEST, mode = READ_WRITE)
`

Reply:
To avert potential problems I'll go for a common lock on all Spring tests